integration/spark/src/main/scala/org/apache/spark/sql/execution/command/carbonTableSchemaCommon.scala

```diff
@@ -235,6 +236,26 @@ class AlterTableColumnSchemaGenerator(
     dataType.toString.equals("STRUCT")
   }
 
+  private def createChildSchema(childField: Field, currentSchemaOrdinal: Int): ColumnSchema = {
+    // TODO: should support adding multi-level complex columns: CARBONDATA-4164
+    if (!childField.children.contains(null)) {
```

Comment:
.contains, can we replace with `isDefined`? thats more clear

Reply:
have put check on size